### ISSUE TITLE
Feature Request: recompute gradient with updated weights within a graph

### ISSUE THREAD

**Issue by linlin2010** [tensorflow/tensorflow#13462]
Hi,

I wonder could there could be some new features to recompute gradients with updated weights within a graph or if there is any better way to do this. For example, for estimating hessian norm, we need to compute

delta ~ N(0, I)
hessian_norm = 1/M \sum_{1}^{M}  gradient(f(x+delta))- gradient(f(x-delta))/(2delta)

we need to gradient value on x+delta. Currently we will get None type if we use tf.gradient on var+delta directly. 

Thank you very much.

**Comment by linlin2010** [comment]
Similar operations can be implemented by pytorch:

https://github.com/ucla-vision/entropy-sgd/blob/master/python/optim.py line63-84. gradients are computed on updated weights within a graph. It might be worthwhile for tensorflow to have similar abilities to do this.

**Comment by yaroslavvb** [comment]
This is too vague for a feature request. It's already possible in tensorflow, StackOverflow is a better place to ask. If you send me a link to your SO question I'll post one possible solution

**Comment by asimshankar** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!

**Comment by lincoln2010** [comment]
https://stackoverflow.com/questions/46672168/how-to-use-tensorflow-to-approximate-hessian-matrixs-norm

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue? Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue? Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by yaroslavvb** [comment]
closing for vagueness
